### PR TITLE
fix shared folders on OS X/Linux, fix composer installation, ...

### PR DESCRIPTION
**Pull request by mirsch**
see http://forum.opencaching.de/index.php?topic=4554.0

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -1,34 +1,73 @@
...
 # vi: set ft=ruby :
 
+VM_IP		= "10.10.0.101"   # IP of the VM, must be unique in your network
+VM_MEMORY	= 512
+VM_CPUS		= 1
+VM_NAME		= "OpenCaching DEV VM"
+VM_HOST		= "local.opencaching.de"
+
+if Vagrant::Util::Platform.windows?
+  HOSTS_FILE = 'c:\WINDOWS\system32\drivers\etc\hosts'
```
Comment (following5) [inline Vagrantfile]: Windows muss nicht auf C: liegen (bei mir isses D:). Kannst du das ändern auf Environment-Variable `SystemRoot` + '\system32\drivers\etc\hosts' ?

Comment (mirsch) [inline Vagrantfile]: Guter Einwand

**Comment by MacGyver-NRW** [comment]
Does not work out-of-the-box on Ubuntu.

In my case, i have to install some libs which wasn't needed before:
» sudo vagrant plugin install vagrant-hostmanager
» sudo vagrant plugin install vagrant-vbguest
» sudo apt-get install nfs-common

nfs is installed but "sudo vagrant up" still throws  this:

> It appears your machine doesn't support NFS, or there is not an
> adapter to enable NFS on this machine for Vagrant. Please verify
> that `nfsd` is installed on your machine, and try again. If you're
> on Windows, NFS isn't supported. If the problem persists, please
> contact Vagrant support.

**Comment by mirsch** [comment]
vagrant-hostmanager is optional and for convenience. It automatically adds or removes local.opencaching.de to your /etc/hosts file. You may add this entry manually if you like. 

I assume you got the "Unknown configuration section 'vbguest'" error. I fixed this in the last commit.

I was not aware that nfsd is not installed by default on Linux (it is on OS X).
Please try
sudo apt-get install nfs-common nfs-kernel-server
Otherwise we can use vboxsf on Linux, but you'll have to adjust file permissions yourself like it was before.

**Comment by teiling88** [comment]
everything seems to work as expected :-)
tested on Ubuntu 16.04

**Comment by MacGyver-NRW** [comment]
@Slini11 | Try on Windows please :)
» OSX | done
» Ubuntu | done

**Comment by teiling88** [comment]
good job @mirsch
@Slini11 tested it on windows 10 and everything worked fine :+1:

**Pull request merged**